### PR TITLE
Improve grouping of pull request template

### PR DESCRIPTION
**Pull request by carlinmack**
:heart: Thank you for your contribution!

Note: this is using my new PR template

### Description

> Please describe briefly your pull request.

I think people sometimes skip ticking the boxes because they see ones which are not applicable and think it's not relevant to them. The changes I've made are the following:

- Group the sections so that the ones which are relevant to all PRs are shown first
- Added a space for people to enter the description

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant documentation.
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've added relevant test cases.

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -2,26 +2,32 @@
...
-- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/) (for relevant code).
-- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
-- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
-- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
 - [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
 - [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
+    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).
+
+**Backend**
```
Comment (ntarocco) [inline .github/pull_request_template.md:20]: I hope we are going to add frontend tests in the future, and there are frontend tests for some modules.
I would merge backend/frontend sections.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -2,26 +2,32 @@
...
+
+- [ ] I've added relevant test cases.
+
+**Frontend**
 
-**Third-party code**
+- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
```
Comment (kpsherva) [inline .github/pull_request_template.md:26]: this also applies to backend in many cases

Comment (carlinmack) [reply .github/pull_request_template.md:26]: :o didn't realise, will move up to the main section

File: .github/pull_request_template.md (hunk 1 above)
Comment (carlinmack) [reply .github/pull_request_template.md:20]: okay will move to main section :)

**Pull request merged**